### PR TITLE
Remove some of the unused dependencies for tests in each module

### PR DESCRIPTION
**Pull request by tonidero**
### Description
We are adding a bunch of dependencies that are unused on each module. This PR removes some of the easier to remove dependencies. I also removed all the dependencies from `library.gradle` since those are imported on every module. I feel that dependencies belong to each module but lmk if you think otherwise.

### CODE REVIEW

File: library.gradle [hunk 1]
```diff
@@ -43,20 +43,6 @@ android {
     }
 }
 
-dependencies {
-    testImplementation 'androidx.test:core:1.4.0'
-    testImplementation 'androidx.test:runner:1.4.0'
-    testImplementation 'androidx.test:rules:1.4.0'
-    testImplementation 'androidx.test.ext:junit:1.1.3'
-    testImplementation 'org.robolectric:robolectric:4.6.1'
```
Comment (tonidero) [inline library.gradle:51]: Note that the versions on the `library.gradle` file were higher in general than the ones on each module. I believe when there is a conflict of versions, gradle will pick the one with the highest version. So I used the higher version libraries for all the modules. These are all test dependencies so it shouldn't affect users in any case

File: build.gradle [hunk 2]
```diff
@@ -6,6 +6,11 @@ buildscript {
     ext.billingVersion = "4.1.0"
     ext.billingClient3Version = "3.0.2"
     ext.lifecycleVersion = "2.3.0-rc01"
+    ext.testLibrariesVersion = "1.4.0"
+    ext.testJUnitVersion  = "1.1.3"
+    ext.robolectricVersion = "4.6.1"
+    ext.mockkVersion = "1.10.0"
+    ext.assertJVersion = "3.13.2"
```
Comment (NachoSoto) [inline build.gradle:13]: 😻

**Review by NachoSoto — COMMENTED** [review]
Nice clean up!

File: library.gradle [hunk 3]
```diff
@@ -43,20 +43,6 @@ android {
     }
 }
 
-dependencies {
```
[the remark this answers is not in the document]
Comment (vegaro) [reply library.gradle:46]: I agree with you 😄

**Review by vegaro — APPROVED** [review]
I love this PR

**Pull request merged**